File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/winforms/samples/helloworldform/vb/HelloWorldForm.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------


Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Namespace Microsoft.Samples.WinForms.VB.HelloWorldForm

    'Do not load this form into the Windows Form Designer.
    'This QuickStart is an example of a simple hand-written
    'form with controls.
    Public Class HelloWorldForm
        Inherits Form

        Private components As Container

        'Declare the buttons using WithEvents so that we can register declarative
        'event handlers
        Private WithEvents button1 as Button

        Private textBox1 As New TextBox

        'When the application is started show the Form
        <STAThread()> Shared Sub Main()
            System.Windows.Forms.Application.Run(New HelloWorldForm())
        End Sub

        Public Sub New()

            MyBase.New

            'Set up the Form
            Me.Text = "Hello Windows Forms World"
            Me.AutoScaleBaseSize = new Size(5, 13)
            Me.ClientSize = new Size(392, 117)

            'Set the minimum form size to the client size + the height of the title bar
            Me.MinimumSize = new Size(392, (117 + SystemInformation.CaptionHeight))

            'Create button1
            button1 = new Button()
            button1.Location = new Point(56, 64)
            button1.Size = new Size(90, 40)
            button1.TabIndex = 2
            button1.Text = "Click Me!"

            'Create the text box
            textBox1.Text = "Hello Windows Forms World"
            textBox1.TabIndex = 1
            textBox1.Size = new Size(360, 20)
            textBox1.Location = new Point(16, 24)

            'Set the default button on the form
            Me.AcceptButton=button1

            'Add the controls to the form
            Me.Controls.Add(button1)
            Me.Controls.Add(textBox1)

        End Sub

        'Called when the Form closes
        'Note that the MessageBox call is simply to
        'demonstrate that Dispose is called. You
        'should keep your Dispose method code as simple and
        'robust as possible
        Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
            Try
                MessageBox.Show("Disposed!")
            Catch ex As Exception
            End Try
            MyBase.Dispose(disposing)
        End Sub



        'The event handling method for button2 - registered using Handles
        Private Sub button1_Click(sender As Object, evArgs As EventArgs) Handles button1.Click

            'Disable button1 - we only want to add one button
            button1.Enabled=False

            'Add the new button and add an event handler using AddHandler
            Dim newButton As new Button
            newButton = new Button()
            newButton.Location = new Point(256, 64)
            newButton.Size = new Size(90, 40)
            newButton.TabIndex = 4
            newButton.Text = "Click Me Too!"
            Me.Controls.Add(newButton)

            AddHandler newButton.Click, AddressOf Me.clickNewbutton
        End Sub

        'The event handling method for the new button -  registered using AddHandler
        Private Sub clickNewbutton(sender As Object, evArgs As EventArgs)
            MessageBox.Show("Hello from the new Button")
        End Sub

    End Class

End Namespace

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/winforms/samples/helloworldform/cs/helloworldform.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------


namespace Microsoft.Samples.WinForms.Cs.HelloWorldForm {
    using System;
    using System.Windows.Forms;
    using System.Drawing;

    // Do not load this form into the Windows Form Designer.
    // This QuickStart is an example of a simple hand-written
    // form with controls.
    public class HelloWorldForm : System.Windows.Forms.Form {

        private Button button1 = new Button() ;
        private TextBox textBox1 = new TextBox();

        //When the application is started show the Form
        [STAThread]
        public static int Main(string[] args) {
            Application.Run(new HelloWorldForm());
            return 0;
        }

        public HelloWorldForm() {

            //Set up the Form
            this.Text = "Hello Windows Forms World";
            this.AutoScaleBaseSize = new Size(5, 13);
            this.ClientSize = new Size(392, 117);

            //Set the minimum form size to the client size + the height of the title bar
            this.MinimumSize = new Size(392, (117 + SystemInformation.CaptionHeight));

            //Set the default button on the Form
            this.AcceptButton=button1;

            //Create the button
            button1.Location = new Point(256, 64);
            button1.Size = new Size(120, 40);
            button1.TabIndex = 2;
            button1.Text = "Click Me!";

            //Register the event handler
            button1.Click += new System.EventHandler(button1_Click);

            //Create the text box
            textBox1.Text = "Hello Windows Forms World";
            textBox1.TabIndex = 1;
            textBox1.Size = new Size(360, 20);
            textBox1.Location = new Point(16, 24);

            //Add the controls to the form
            this.Controls.Add(button1);
            this.Controls.Add(textBox1);
        }

        //Called when the Form closes
        //Note that the MessageBox call is simply to
        //demonstrate that Dispose is called. You
        //should keep your Dispose method code as simple and
        //robust as possible
        protected override void Dispose(bool disposing)
        {
            try {
                MessageBox.Show("Disposed!");
            } catch(Exception) {}
            base.Dispose(disposing);
        }

        //The event handling method
        private void button1_Click(object sender, EventArgs evArgs) {
            MessageBox.Show("Text is: '" + textBox1.Text + "'");
        }
    }
}









